### DISCUSSION TITLE
Using `workflow_run` in pull requests from non-default branches or forks

### DISCUSSION THREAD

**Discussion by yarnabrina** [community/community#72097] [Actions] [answered]
### Select Topic Area

Question

### Body

I'm working a python repository, where I am using `pre-commit`. I have one workflow that runs code quality checks, and a set of other workflows (>=1) that will run different types of unit tests.

I do not wish to trigger the testing workflows unless code quality cheks pass. I've set up configuration using `workflow_run` but it is working only on default branch `main`. In pull requests, only code quality workflow gets triggered and not the testing workflows.

Example configurations from a sample repository:

1. code quality workflow: https://github.com/yarnabrina/learn-python-packaging/blob/0161af15b6ee899178a66951def2bf06e5be0ea8/.github/workflows/pre-commit.yml
2. test workflow: https://github.com/yarnabrina/learn-python-packaging/blob/0161af15b6ee899178a66951def2bf06e5be0ea8/.github/workflows/test.yml

If someone can suggest what am I doing wrong, that'd be great. Thanks.

**Comment by mickeygousset** [comment, marked as answer]
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run

The workflow_run event only works on the default branch.

instead of using workflow_run, look at using reusable workflows.  Then in your code quality workflow, if everything works there, you would call a reusable workflow to do the testing.

**Comment by AlexSkrypnyk** [comment]
> The workflow_run event only works on the default branch.

This is not exactly what the documentation says:

> This event will only trigger a workflow run if the workflow file exists on the default branch.

It is quite confusing because it reads as the file has to exist on the default branch, but there is nothing about on which branch the _triggering_ workflow can be.

In the OP case, the triggering workflow is on the feature branch. As per the description above, the workflow should be triggered in this case.

**Reply by Hotell** [reply]
+1 on @AlexSkrypnyk , we ran into this confusing behaviours as well. it is necessary in order to do various CI  checks on feature/experimental branches.


this is our situation 

<img width="1089" alt="image" src="https://github.com/user-attachments/assets/6984ec69-2dab-48cf-8c9d-f2817bd0d3d9" />

ty

**Reply by Hotell** [reply]
_update:_ once we removed the `branches` filter from `workflow_run` it works as expected.

**Reply by layonez** [reply]
@Hotell is the job status reported for you in PR checks?
